File: app/src/main/java/com/hillal/hhhhhhh/data/local/dao/AccountDao.java
package com.hillal.hhhhhhh.data.local.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;
import com.hillal.hhhhhhh.data.local.entity.Account;
import java.util.List;

@Dao
public interface AccountDao {
    @Query("SELECT * FROM accounts")
    List<Account> getAllAccounts();

    @Query("SELECT * FROM accounts WHERE id = :id")
    Account getAccountById(long id);

    @Query("SELECT * FROM accounts WHERE server_id = :serverId")
    Account getAccountByServerId(long serverId);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insert(Account account);

    @Update
    void update(Account account);

    @Delete
    void delete(Account account);

    @Query("DELETE FROM accounts")
    void deleteAll();

    @Query("SELECT * FROM accounts WHERE user_id = :userId")
    List<Account> getAccountsByUserId(long userId);
} 

File: android_app/app/src/main/java/com/hsaby/accounting/data/local/dao/AccountDao.kt
package com.hsaby.accounting.data.local.dao

import androidx.room.*
import com.hsaby.accounting.data.local.entity.AccountEntity
import kotlinx.coroutines.flow.Flow

@Dao
interface AccountDao {
    @Query("SELECT * FROM accounts")
    fun getAllAccounts(): Flow<List<AccountEntity>>

    @Query("SELECT * FROM accounts WHERE userId = :userId")
    fun getAccountsByUserId(userId: String): Flow<List<AccountEntity>>

    @Query("SELECT * FROM accounts WHERE isSynced = 0")
    fun getUnsyncedAccounts(): Flow<List<AccountEntity>>

    @Query("SELECT * FROM accounts WHERE isActive = 1")
    fun getActiveAccounts(): Flow<List<AccountEntity>>

    @Query("SELECT * FROM accounts WHERE isDebtor = 1")
    fun getDebtorAccounts(): Flow<List<AccountEntity>>

    @Query("SELECT * FROM accounts WHERE id = :id")
    suspend fun getAccountById(id: String): AccountEntity?

    @Query("SELECT * FROM accounts WHERE serverId = :serverId")
    suspend fun getAccountByServerId(serverId: String): AccountEntity?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertAccount(account: AccountEntity)

    @Update
    suspend fun updateAccount(account: AccountEntity)

    @Delete
    suspend fun deleteAccount(account: AccountEntity)

    @Query("DELETE FROM accounts WHERE userId = :userId")
    suspend fun deleteAllAccounts(userId: String)

    @Query("UPDATE accounts SET serverId = :serverId WHERE id = :localId")
    suspend fun updateServerId(localId: Long, serverId: String)

    @Query("UPDATE accounts SET balance = :newBalance WHERE id = :accountId")
    suspend fun updateAccountBalance(accountId: Long, newBalance: Double)

    @Query("UPDATE accounts SET isActive = :isActive WHERE id = :accountId")
    suspend fun updateAccountStatus(accountId: Long, isActive: Boolean)

    @Query("SELECT * FROM accounts WHERE accountName LIKE :query OR phoneNumber LIKE :query")
    fun searchAccounts(query: String): Flow<List<AccountEntity>>

    @Query("SELECT SUM(balance) FROM accounts WHERE isDebtor = 0")
    suspend fun getTotalBalance(): Double?

    @Query("SELECT SUM(balance) FROM accounts WHERE isDebtor = 1")
    suspend fun getTotalDebtors(): Double?

    @Query("SELECT COUNT(*) FROM accounts WHERE isActive = 1")
    suspend fun getActiveAccountsCount(): Int
} 